### ISSUE TITLE
How can I use tf.data.experimental.make_csv_dataset to make a dataset with no batches

### ISSUE THREAD

**Issue by aflah02** [tensorflow/tensorflow#55261]
Currently I'm doing this - 

```
column_names = ['Label','Sentence']
batchsize = 32
label = column_names[0]
train_dataset = tf.data.experimental.make_csv_dataset(
    'datasettrain.csv',
    batchsize,
    column_names = column_names,
    label_name = label,
    num_epochs=1
)
```

Due to this the batches being ordered dicts don't allow me to do certain things which a dataset directly loaded from tfds.load would allow. For instance in this tutorial - https://www.tensorflow.org/text/tutorials/text_classification_rnn

The author is able to run - 

```
for example, label in train_dataset.take(1):
  print('text: ', example.numpy())
  print('label: ', label.numpy())
```

and I wish to do the same however in my case the type for example comes out to be `<class 'collections.OrderedDict'>` while in the tutorials it is `<class 'tensorflow.python.framework.ops.EagerTensor'>`.

Is there some way I can do the same with `tf.data.experimental.make_csv_dataset`?

Using -

```
ls = []
for example, label in train_dataset:
  ls.append([example['Sentence'][0],label[0]])
```

I was able to get the required form internally however the entire array is no longer a `<class 'tensorflow.python.data.ops.dataset_ops.PrefetchDataset'>` is there any way to do that from this `ls`?

**Comment by mohantym** [comment]
Hi @aflah02 ! Can you please share the code which you were trying to execute as a Colab gist?

**Comment by aflah02** [comment]
Hey
The code which is causing the error is the 
one mentioned in the issue already
Which 2 methods are you referring to?

**Comment by aflah02** [comment]
Incase some one faces a similar issue I eventually got a workaround by making 2 numpy arrays converting them into tensors and then feeding that into tf.data.Dataset.from_tensors but I guess there would be a better way than this the problem seems to be that Tensorflow does not recognize a np array of EagerTensors as a valid argument for making a dataset

**Comment by mohantym** [comment]
Ok @aflah02 ! Glad it resolved. Could you please share the resolved code for community ?

**Comment by aflah02** [comment]
Sure @mohantym Thanks for pointing it out

For perspective my dataset is of the following form - 

- A CSV
- 2 Columns
- First Column Which I Need to Predict Named Label
- Second Column Which I Use to Predict the Label which is a Sentence

So I first loaded it in using - 

```
column_names = ['Label','Sentence']
batchsize = 1
label = column_names[0]
train_dataset = tf.data.experimental.make_csv_dataset(
    'datasettrain.csv',
    batchsize,
    column_names = column_names,
    label_name = label,
    num_epochs=1
)
```

This returns a `tensorflow.python.data.ops.dataset_ops.PrefetchDataset`

After that I used the following code to do the work mentioned in [Comment](https://github.com/tensorflow/tensorflow/issues/55261#issuecomment-1070543375) above

```
ls_train = []
for example, label in train_dataset:
  ls_train.append([example['Sentence'][0],label[0]])

nparrtrainsentences = np.array([])
nparrtrainlabels = np.array([])
for i in range(len(ls_train)):
  nparrtrainsentences = np.append(nparrtrainsentences, [ls_train[i][0].numpy()])
  nparrtrainlabels = np.append(nparrtrainlabels, [ls_train[i][1].numpy()])

train_sentences = tf.convert_to_tensor(nparrtrainsentences)
train_labels = tf.convert_to_tensor(nparrtrainlabels)

train_dataset = tf.data.Dataset.from_tensors((train_sentences,train_labels))
```

In hindsight certain aspects do feel redundant now but since it works for now I won't be trying to fix those things

**Comment by caesarjuly** [comment]
My friend, actually you can simply add an `.unbatch()` after the `make_csv_dataset`
